main/auth/auth.py

```diff
@@ -132,6 +132,24 @@ def decorated_function(*args, **kwargs):
   return decorated_function
 
 
+def cron_required(f):
+  decorator_order_guard(f, 'auth.cron_required')
+
+  @functools.wraps(f)
+  def decorated_function(*args, **kwargs):
+    if config.PRODUCTION and 'X-Appengine-Cron' in flask.request.headers:
+      return f(*args, **kwargs)
+    if is_logged_in() and current_user_db().admin:
+      return f(*args, **kwargs)
+    if not is_logged_in() and flask.request.path.startswith('/api/'):
+      return flask.abort(401)
+    if not is_logged_in():
+      return flask.redirect(flask.url_for('signin', next=flask.request.url))
+    return flask.abort(403)
+
+  return decorated_function
+
+
```

Comment:
Feels like there should be two endpoints; one for admins to start calculations manually (using `admin_required`) and another one that uses (a simplified) `cron_required` for cron only (perhaps in non-PRODUCTION allow admins for debugging/testing). In that setup, the end-points just call upon a worker method: basically the current `admin_stats_calc` method, but perhaps moved (in its generic form) to `task.py` under Stats Related.

The cron end-point may not take parameters, as `when=yesterday` would probably be the only arguments anyway. It would call the generic method based on values from `now - 1 day`.